test/core/end2end/fuzzers/api_fuzzer.cc

```diff
@@ -407,18 +422,30 @@ class Call : public std::enable_shared_from_this<Call> {
                 : nullptr;
       } break;
       case api_fuzzer::BatchOp::kReceiveInitialMetadata:
-        op.op = GRPC_OP_RECV_INITIAL_METADATA;
-        *batch_ops |= 1 << GRPC_OP_RECV_INITIAL_METADATA;
-        op.data.recv_initial_metadata.recv_initial_metadata =
-            &recv_initial_metadata_;
+        if (enqueued_recv_initial_metadata_) {
+          *batch_is_ok = false;
+        } else {
+          enqueued_recv_initial_metadata_ = true;
+          op.op = GRPC_OP_RECV_INITIAL_METADATA;
+          *batch_ops |= 1 << GRPC_OP_RECV_INITIAL_METADATA;
+          op.data.recv_initial_metadata.recv_initial_metadata =
+              &recv_initial_metadata_;
+        }
         break;
       case api_fuzzer::BatchOp::kReceiveMessage:
-        if (call_closed_) {
+        // Allow only one active pending_recv_message_op to exist. Otherwise if
+        // the previous enqueued recv_message_op is not complete by the time
+        // we get here, then under certain conditions, enqueing this op will
+        // over-write the internal call->receiving_buffer maintained by grpc
+        // leading to a memory leak.
+        if (call_closed_ || pending_recv_message_op_) {
```

Comment:
Is this still true?

Reply:
Yes I think it is. The previously allocated internal buffer needs to be destroyed before it can be over-written. Since the fuzzer is not consuming the received message, we need to do this when the batch is complete.